In [1]:
import pandas as pd

In [2]:
df=pd.read_csv("airbnb_data.csv")

In [3]:
df.head()

id                                               name  room_type  price  \
0   3781                          HARBORSIDE-Walk to subway  Apartment    146   
1   5506     **$49 Special **   Private! Minutes to center!  Apartment    145   
2   6695                     $99 Special!! Home Away! Condo  Apartment    169   
3  10730                     Bright 1bed facing Golden Dome  Apartment     85   
4  10813  Back Bay Apt-blocks to subway, Newbury St, The...  Apartment     87   

   minimum_nights  number_of_reviews    city  
0              28                 16  Boston  
1               3                107  Boston  
2               3                115  Boston  
3              91                 32  Boston  
4              28                 10  Boston

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32348 entries, 0 to 32347
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 32348 non-null  int64 
 1   name               32348 non-null  object
 2   room_type          32348 non-null  object
 3   price              32348 non-null  int64 
 4   minimum_nights     32348 non-null  int64 
 5   number_of_reviews  32348 non-null  int64 
 6   city               32348 non-null  object
dtypes: int64(4), object(3)
memory usage: 1.7+ MB


In [5]:
df.city.value_counts()

Hawaii     22434
Seattle     6575
Boston      3339
Name: city, dtype: int64

In [13]:
df.room_type.value_counts()

Apartment    27241
Private       4660
HotelRoom      272
Shared         175
Name: room_type, dtype: int64

In [7]:
df.isnull().any()

id                   False
name                 False
room_type            False
price                False
minimum_nights       False
number_of_reviews    False
city                 False
dtype: bool

In [16]:
#sample space
ss_ab=df.id.count()
ss_ab

32348

In [15]:
#number of rooms in Boston
series1=df.apply(lambda x:True if x['city']=='Boston' else 'False',axis=1)
bos=len(series1[series1==True])
print(bos)

3339


In [19]:
#probability of finding a room in Boston
p_bos=round(bos/ss_ab,3)
print(p_bos)

0.103


In [20]:
#number of rooms in Hawaii
series2=df.apply(lambda x:True if x['city']=='Hawaii' else 'False',axis=1 )
haw=len(series2[series2==True])
print(haw)

22434


In [22]:
#probability of finding a room in Hawaii
p_haw=round(haw/ss_ab,3)
print(p_haw)


0.694


In [23]:
series3=df.apply(lambda x:True if x['city']=='Seattle' else 'False',axis=1)
sea=len(series3[series3==True])
print(sea)

6575


In [24]:
p_sea=round(sea/ss_ab,3)
print(p_sea)

0.203


# INFERENCES

(1)All three are mutually exclusive events i.e. no tw can occour at the same time .The sum of their probabilities is one.
(2)Probabilities of finding a room in Boston ,Hawaii and Seattle are 0.103 ,0.694 and 0.203 respectively .

# Finding a room either in Boston or Hawaii

In [25]:
p_bos_or_haw=p_bos+p_haw
print(p_bos_or_haw)

0.7969999999999999


# Probability of finding a room in Boston or an Apartment.

In [28]:
n_apt_bos=df[(df['room_type']=='Apartment')&(df['city']=='Boston')].count()['id']
print(n_apt_bos)

2163


0


In [38]:
#number of Apartment type rooms
series4 = df.apply(lambda x: True if x['room_type'] == 'Apartment' else 'False' , axis=1)
apts = len(series4[series4 == True])
print(apts)

27241


In [39]:
p_apts=round(apts/ss_ab,3)
print(p_apts)

0.842


In [40]:
p_aptANDboston=round(n_apt_bos/ss_ab,3)
p_aptANDboston

0.067

In [41]:
p_bosORapt = p_bos + p_apts - p_aptANDboston
p_bosORapt

0.8779999999999999

In [43]:
n_sea_pvt=df[(df['city']=='Seattle')&(df['room_type']=='Private')].count()['id']
n_sea_pvt

1399

In [44]:
p_pvtANDsea=round(n_sea_pvt/ss_ab,3)
p_pvtANDsea

0.043